macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +234,402 @@ open class Button: NSButton {
-			var verticalInterCellSpacing: CGFloat = 0
-			switch imagePosition {
-			
-			case .noImage:
-				break
-			case .imageOnly:
-				break
-			case .imageLeft:
-				horizontalInterCellSpacing = Button.titleAndImageRectInterspacing
-				break
-			case .imageRight:
-				horizontalInterCellSpacing = Button.titleAndImageRectInterspacing
-				break
-			case .imageBelow:
-				verticalInterCellSpacing = Button.titleAndImageRectInterspacing
-				break
-			case .imageAbove:
-				verticalInterCellSpacing = Button.titleAndImageRectInterspacing
-				break
-			case .imageOverlaps:
-				break
-			case .imageLeading:
-				horizontalInterCellSpacing = Button.titleAndImageRectInterspacing
-				break
-			case .imageTrailing:
-				horizontalInterCellSpacing = Button.titleAndImageRectInterspacing
-				break
-			@unknown default:
-				break
-			}
+		return NSMakeRect(x, y, width, height)
+	}
 
-			let drawingRectWithPadding = NSMakeRect(
-				0,
-				0,
-				rect.width - (Button.horizontalEdgePadding * 2) - horizontalInterCellSpacing,
-				rect.height - (Button.verticalEdgePadding * 2) - verticalInterCellSpacing
-			)
-			return drawingRectWithPadding
+	override func drawingRect(forBounds rect: NSRect) -> NSRect {
+		var horizontalInterCellSpacing: CGFloat = 0
+		var verticalInterCellSpacing: CGFloat = 0
+
+		switch imagePosition {
+		case .imageLeft, .imageLeading, .imageRight, .imageTrailing:
+			horizontalInterCellSpacing = titleToImageSpacing
+		case .imageBelow, .imageAbove:
+			verticalInterCellSpacing = titleToImageSpacing
+		case .noImage, .imageOnly, .imageOverlaps:
+			break
+		@unknown default:
+			break
 		}
+
+		let drawingRectWithPadding = NSMakeRect(
+			0,
+			0,
+			rect.width - (horizontalPadding * 2) - horizontalInterCellSpacing,
+			rect.height - (verticalPadding * 2) - verticalInterCellSpacing
+		)
+		return drawingRectWithPadding
 	}
+}
 
-	private static let titleAndImageRectInterspacing: CGFloat = 8
+// MARK: - Button Formatting Parameters
 
+/// Indicates the size of the button
+@objc(MSFButtonSize)
+public enum ButtonSize: Int, CaseIterable {
+	case large
+	case small
+}
 
-	private static let borderWidth: CGFloat = 1
+/// Indicates what style our button is drawn as
+@objc(MSFButtonStyle)
+public enum ButtonStyle: Int, CaseIterable {
+	/// Accent color fill, white text/image.
+	case primary
 
-	private static let cornerRadius: CGFloat = 3
+	/// Light mode: white fill, black text/image, light gray outline.
+	/// Dark mode: dark gray fill, white text/image, subtle gray outline.
+	/// Pressed: accent color fill, white text/image, no outline (same as primary style).
+	case secondary
+
+	/// Light mode: light gray fill, black text/image.
+	/// Dark mode: dark gray fill, white text/image.
+	case acrylic
+
+	/// Accent color text/image, no fill or outline.
+	case borderless
+}
 
-	private static let verticalEdgePadding: CGFloat = 4
+/// Combination of all formatting parameters.
+public struct ButtonFormat {
```

Comment:
do we need this struct?

Reply:
There's already another thread on this subject.  I made the struct because it's convenient to create a handful of format structs and apply them to buttons en masse using the designated initializer that accepts this struct (or using the synthesized `format` property.  I'm using it extensively in the test app.